packages/clay-core/src/tree-view/TreeView.tsx

```diff
@@ -24,6 +24,7 @@ interface ITreeViewProps<T>
 	expanderIcons?: Icons;
 	onLoadMore?: (item: T) => Promise<unknown>;
 	onRenameItem?: (item: T) => Promise<T>;
+	multipleSelection?: boolean;
```

Comment:
@julien I think we can change the name to say that it won't actually disable multi-selection but change the selection type, for example something like:
```
selectionMode: 'multiple' | 'single'
```

But still, I think that could be another name 🤔 I think it's still confusing for the behavior, we have to have an API to say we're not doing recursive selection, maybe `recursiveSelection`.

Reply:
I actually like `selectionMode: 'multiple' | 'single'`, and I think that if we used `recursiveSelection` that would imply that it's a `boolean`. We can keep on talking about this.

Reply:
Yeah, I think if we go with `selectionMode` with the type `multiple` and `single` it might cause a bit of confusion after I thought about it, for example, `single` is more related that it's only allowed to select one node only in every tree.

Reply:
@matuzalemsteles I think `selectionMode="single"` means that you can select one node at a time. I'm happy with this as it is.

Reply:
Hmm, I always have the intuition that a `selection`, single is related I can just select a Node in the TreeView, for example the `select` tag by default it is a single and you can enable multiple selection of multiple nodes by defining `multiple="multiple"`.

I also find that a single is more intuitive for: I can only select one Node than selecting one at a time, this is also similar to how the user interacts on a form, a Single Selection renders Radios which just means selecting one item within the group and Multiple Selection will render a checkbox which is the multiple selection.

I think the problem here is that this may sound in another sense of what he really wants to do and it can cause some confusion. I'm not saying this is wrong just to try to avoid the API's double meaning.

Reply:
You might be right, I think if we specify this in the component's documentation it will make it easier for others to understand, but of course we can rename it if you prefer. I'm OK with these last changes.